### DISCUSSION TITLE
关于正文中标题左对齐但摘要目录参考文献等标题居中的问题

### DISCUSSION THREAD

**Discussion by favefan** [sjtug/SJTUThesis#1028] [提问]
请问如何实现正文的章标题左对齐显示，但是摘要目录参考文献等标题居中显示呢？在 [SJTUTeX](https://github.com/sjtug/SJTUTeX) 中实验多次未果，要么就是标题全都一起居中要么就是全都左对齐。感谢！希望的效果如下：
![image](https://github.com/sjtug/SJTUThesis/assets/25415931/7e98cf44-d153-41b0-9855-500a97b8402b)
![image](https://github.com/sjtug/SJTUThesis/assets/25415931/9dd2afe0-00af-4d9d-9174-5a9a9dbbe2e2)
![image](https://github.com/sjtug/SJTUThesis/assets/25415931/f60e886b-ac72-4611-b3c0-737e5df18c7d)

**Comment by AlexaraWu** [comment]
环境的内部是一个单独的作用域，在环境定义里设置章标题居中就可以了：
```latex
\NewDocumnetEnvironment{abstract}{}
  {
    \ctexset{chapter/format+=\centering}
    ...
  }
  {
    ...
  }
```

**Reply by favefan** [reply]
那么对于在 `\DeclareDocumentCommand` 中所定义目录部分的标题，进行单独的居中设置，您有什么建议吗？

```
\DeclareDocumentCommand \tableofcontents { s }
  {
    \IfBooleanTF {#1}
      {
        \SJTU@head* { \contentsname } 
      }
      {
        \SJTU@head  { \contentsname } 
      }
    \group_begin:
      \cs_set:Npn \makebox [##1][##2]##3 { \, ##3 }
      \@starttoc { toc }
    \group_end:
  }
```

![image](https://github.com/sjtug/SJTUThesis/assets/25415931/9dd2afe0-00af-4d9d-9174-5a9a9dbbe2e2)

**Reply by AlexaraWu** [reply]
`\group_begin:` 与 `\group_end:` 分组之内也是一个单独的作用域。

**Comment by favefan** [comment]
可以了，谢谢！

在 2024年3月31日，16:25，Alexara Wu ***@***.***> 写道：

﻿

环境的内部是一个单独的作用域，在环境定义里设置章标题居中就可以了：

\NewDocumnetEnvironment{abstract}{}
  {
    \ctexset{chapter/format+=\centering}
    ...
  }
  {
    ...
  }
